lib/src/widgets/chatui_textfield.dart

```diff
@@ -36,48 +36,121 @@ class ChatUITextField extends StatelessWidget {
+                      color: Colors.grey.shade600,
+                      letterSpacing: 0.25,
+                    ),
+                contentPadding: const EdgeInsets.symmetric(horizontal: 6),
+                border: _outLineBorder,
+                focusedBorder: _outLineBorder,
+                enabledBorder: OutlineInputBorder(
+                  borderSide: const BorderSide(color: Colors.transparent),
+                  borderRadius: BorderRadius.circular(29),
+                ),
+              ),
             ),
-        contentPadding: const EdgeInsets.symmetric(
-          vertical: 6,
-          horizontal: 14,
-        ),
-        border: _outLineBorder,
-        focusedBorder: _outLineBorder,
-        enabledBorder: OutlineInputBorder(
-          borderSide: const BorderSide(color: Colors.transparent),
-          borderRadius: BorderRadius.circular(29),
-        ),
+          ),
+          _inputText.isNotEmpty
+              ? IconButton(
+                  color:
+                      sendMessageConfig?.defaultSendButtonColor ?? Colors.green,
+                  onPressed: widget.onPressed,
+                  icon: sendMessageConfig?.sendButtonIcon ??
+                      const Icon(Icons.send),
+                )
+              : Row(
+                  children: [
+                    IconButton(
+                      constraints: const BoxConstraints(),
+                      onPressed: () => _onIconPressed(
+                        imagePickerIconsConfig?.onCameraIconPressed,
+                        ImageSource.camera,
+                      ),
+                      icon: imagePickerIconsConfig?.cameraImagePickerIcon ??
+                          Icon(
+                            Icons.camera_alt_outlined,
+                            color: imagePickerIconsConfig?.cameraIconColor,
+                          ),
+                    ),
+                    IconButton(
+                      constraints: const BoxConstraints(),
+                      onPressed: () => _onIconPressed(
+                        imagePickerIconsConfig?.onGalleryIconPressed,
+                        ImageSource.gallery,
+                      ),
+                      icon: imagePickerIconsConfig?.galleryImagePickerIcon ??
+                          Icon(
+                            Icons.image,
+                            color: imagePickerIconsConfig?.galleryIconColor,
+                          ),
+                    ),
+                  ],
+                ),
+        ],
       ),
     );
   }
+
+  void _onIconPressed(
+      StringCallback? onIconPressed, ImageSource imageSource) async {
+    try {
+      if (onIconPressed != null) {
+        final XFile? _image = await _imagePicker.pickImage(source: imageSource);
+        onIconPressed(_image?.path ?? '');
+      }
+    } catch (e) {
+      if (imagePickerIconsConfig?.onImagePickerError != null) {
+        imagePickerIconsConfig?.onImagePickerError!(e.toString());
```

Comment:
instead of separate function you can use the same function on onImageSelected. and add extra parameter error in it.

Reply:
Updated